src/index/Index.Text.cpp

```diff
@@ -10,23 +12,109 @@
 
 #include "../engine/CallFixedSize.h"
 #include "../parser/ContextFileParser.h"
+#include "../util/Conversions.h"
 #include "../util/Simple8bCode.h"
 #include "./FTSAlgorithms.h"
 #include "./Index.h"
 
+// _____________________________________________________________________________
+cppcoro::generator<ContextFileParser::Line> Index::contextFileLines(
+    const std::string& contextFile, bool useRdfVocabularyAsText) {
+  auto localeManager = _textVocab.getLocaleManager();
+  // ROUND 1: Read from context file aka wordsfile if not empty. Remember
+  // the last context id for the (optional) second round.
+  //
+  // TODO: For now, EITHER do this round OR the other, see the respective
+  // comment in `addTextFromContextFile` below.
+  TextRecordIndex contextId = TextRecordIndex::make(0);
+  if (!useRdfVocabularyAsText) {
+    ContextFileParser::Line line;
+    ContextFileParser p(contextFile, localeManager);
+    ad_utility::HashSet<string> items;
+    while (p.getLine(line)) {
+      contextId = line._contextId;
+      co_yield line;
+    }
+    if (contextId > TextRecordIndex::make(0)) {
+      contextId = contextId.incremented();
+    }
```

Comment:
1. Does the ContextFileParser increment the `_contextId` automatically? Maybe this could be commented.
2. If I read it correctly, either the first half or the second half of this function are executed, which makes this increment a noop. Can't the two rounds be executed at once?

Reply:
@1: The `ContextFileParser` reads the context ids from the wordsfile. All our wordsfiles have context ids in ascending order starting from 0. I'm not sure what happens if they don't (my guess: it still works, just less efficiently so because of the gap encoding).

@2: You are right that in the current version of the code, *either* Round 1 *or* Round 2 are executed, never both. But I realized when writing this code that it's no problem to have both rounds, we just need an option to activate it. Then we can first read the text records from the wordsfile and afterwards create *additional* records from (a subset of) the literals and IRIs. To prepare for that, I added the lines on which you commented. This is also documented in a comment a few lines further up.

Incidentally: I used the pattern `someIndex = someIndex.incremented()` so often that I now think that a `++` operator or at least an `increment` function would be good to have.